File: 考研计算机/数据结构/code/ds/_8_algo/练习/对撞指针技巧及其代码模版/CollisionPointer.js
// 数组反转
function reverse(nums) {
    let left = 0, right = nums.length - 1;
    while (left < right) {
        let tmp = nums[left];
        nums[left] = nums[right];
        nums[right] = tmp;

        left++;
        right--;
    }
}

// 验证回文串
function isPalindrome(s) {
    let left = 0, right = s.length - 1;
    while (left < right) {
        if (s[left] !== s[right]) {
            return false;
        }
        left++;
        right--;
    }
    return true;
}

File: 考研计算机/数据结构/code/ds/_8_algo/练习/对撞指针技巧及其代码模版/CollisionPointer.cpp
// 数组反转
void reverse(vector<int>& nums){
    int left = 0, right = nums.size() - 1;
    while (left < right) {
        int tmp = nums[left];
        nums[left] = nums[right];
        nums[right] = tmp;

        left++;
        right--;
    }
}

#include <string>

using namespace std;
// 验证回文串
bool isPalindrome(string s) {
    int left = 0, right = s.length() - 1;
    while (left < right) {
        if (s[left] != s[right]) {
            return false;
        }
        left++;
        right--;
    }
    return true;
}